### PR TITLE
Add color picker

### PR DESCRIPTION
**Pull request by xethlyx**
image
Color picker!

### CODE REVIEW

File: src/util/colorMap.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	new = 'new',
+	fromRGB = 'fromRGB',
+	fromHSV = 'fromHSV'
+}
+
+type ExcludeSelfDoubleRecord<K extends string, V> = { [I in K]: Record<Exclude<K, I>, V> };
+export type ColorArray = [number, number, number];
+
+function clearNaN(value: Readonly<ColorArray>) {
+	return value.map(value => value !== value ? 0 : value) as ColorArray;
```
Comment (osyrisrblx) [inline src/util/colorMap.ts:14]: Add a return type for the function `: ColorArray`, safer than doing a type assertion. (Also the next two functions as well)

**Review by osyrisrblx — APPROVED** [review]

File: src/colorizePrint.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		provideColorPresentations: (color, context, token) => {
+			const text = context.document.getText(context.range);
+
+			const match = Object.values(ColorType).find(match => text.includes(match));
+			if (!match) throw new Error('Color type specified was not found!');
+
+			const regexMatch = text.match(matchColors[match])!;
+
+			const matches: vscode.ProviderResult<Array<vscode.ColorPresentation>> = Object.entries(ColorType).map(([matchTypeRaw]): vscode.ColorPresentation => {
+				const matchType = matchTypeRaw as ColorType;
```
Comment (Fireboltofdeath) [inline src/colorizePrint.ts]: Is there a reason for the second variable and type cast here? Couldn't you just set the parameter as `[, matchType]` to grab the ColorType?

Comment (xethlyx) [reply src/colorizePrint.ts]: No there isn't, I'll remove that

Comment (xethlyx) [reply src/colorizePrint.ts]: On second thought, there isn't a need for entries anyways, I'll replace that with values

**Pull request merged**